monitoring/uss_qualifier/scd/executor/executor.py

```diff
@@ -73,17 +76,53 @@ def format_combination(combination: Dict[str, TestTarget]) -> List[str]:
 
 
 def run_scd_tests(locale: Locality, test_configuration: SCDQualifierTestConfiguration,
-                  auth_spec: str):
+                  auth_spec: str) -> bool:
     automated_tests = load_scd_test_definitions(locale)
     configured_targets = list(map(lambda t: TestTarget(t.name, t, auth_spec), test_configuration.injection_targets))
+    report = Report(
+            qualifier_version=os.environ.get("SCD_VERSION", "unknown"),
+            configuration=test_configuration,
+    )
 
+    should_exit = False
+    executed_test_run_count = 0
     for test_id, test in automated_tests.items():
+        if should_exit:
+            break
         target_combinations = combine_targets(configured_targets, test.steps)
         for i, targets_under_test in enumerate(target_combinations):
+            context = AutomatedTestContext(
+                test_id = test_id,
+                test_name = test.name,
+                locale = locale,
+                targets_combination = dict(map(lambda t: (t[0], t[1].name), targets_under_test.items()))
+            )
             print('[SCD] Starting test combination {}: {} ({}/{}) {}'.format(i+1,  test.name, locale, test_id,
                 format_combination(targets_under_test)))
 
-            runner = TestRunner(test_id, test, targets_under_test)
+            runner = TestRunner(context, test, targets_under_test, report)
+            try:
+                runner.run_automated_test()
+            except TestRunnerError as e:
+                print("[SCD] TestRunnerError: {} Issue: {} Related interactions: {}".format(e, e.issue.details, e.issue.interactions))
+            finally:
+                runner.teardown()
+
+            executed_test_run_count = executed_test_run_count + 1
+            should_exit = len(report.findings.critical_issues()) > 0
+            if should_exit:
+                print ("[SCD] Critical issues found during test. Interrupting test sequence. {}".format(report.findings))
+                break
```

Comment:
We may want to consider what is still doable under what failures.  A critical failure of one test will certainly stop that test, but other tests may still be possible (we do expect to be able to re-run this entire executable after all).

Reply:
Can we agree on the following severities description?
- **Critical**: Error is unrecoverable and left the system dirty preventing subsequent correct test run execution. For instance, inability to clean a created flight which will conflict with another run with a different combination of targets.
- **High**: Error interrupts a test run and doesn't impact subsequent test runs. For instance, the test failed and all created can safely be clean up during teardown by the test driver.
- **Low**: Error doesn't impact the test run and further test steps can be executed without impact.

Reply:
Yes, those look good to me.  I'd expect we would augment the [existing descriptions](https://github.com/interuss/dss/blob/067e550199182e59b447843395f1faee488d07cc/monitoring/uss_qualifier/common_data_definitions.py#L7) with these descriptions.  I'd also suggest **Medium**: The system does not meet requirements, but further test steps can likely be executed without impact.

Reply:
#720